activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/ExtensionsModelValidator.java

```diff
@@ -15,26 +15,231 @@
+                .map(valiableMappingEntry -> validateProcessVariableMapping(action,
+                                                                            valiableMappingEntry.getKey(),
+                                                                            valiableMappingEntry.getValue(),
+                                                                            modelId,
+                                                                            availableProcessVariables))
+                .filter(Optional::isPresent)
+                .map(Optional::get);
+    }
+
+    private Optional<ModelValidationError> validateTaskName(String taskName,
+                                                            String modelId,
+                                                            Set<String> availableTasks) {
+        return !availableTasks.contains(taskName) ?
+                Optional.of(createModelValidationError(
+                        format(UNKNOWN_TASK_VALIDATION_ERROR_PROBLEM,
+                               taskName),
+                        format(UNKNOWN_TASK_VALIDATION_ERROR_DESCRIPTION,
+                               modelId,
+                               taskName))) :
+                Optional.empty();
+    }
+
+    private Optional<ModelValidationError> validateProcessVariableMapping(ServiceTaskActionType action,
+                                                                          String processVariableMappingKey,
+                                                                          ProcessVariableMapping processVariableMapping,
+                                                                          String modelId,
+                                                                          Set<String> availableProcessVariables) {
+        String variableName = action == ServiceTaskActionType.INPUTS ?
+                processVariableMapping.getValue() :
+                processVariableMappingKey;
+        return processVariableMapping.getType() == VARIABLE &&
+                !availableProcessVariables.contains(variableName) ?
+                Optional.of(createModelValidationError(
+                        format(UNKNOWN_PROCESS_VARIABLE_VALIDATION_ERROR_PROBLEM,
+                               variableName),
+                        format(UNKNOWN_PROCESS_VARIABLE_VALIDATION_ERROR_DESCRIPTION,
+                               modelId,
+                               variableName))) :
+                Optional.empty();
+    }
+
+    private Optional<Model> findModelInContext(String modelId,
+                                               ValidationContext validationContext) {
+        return validationContext.getAvailableModels()
+                .stream()
+                .filter(model -> Objects.equals(model.getId(),
+                                                modelId))
+                .findFirst();
+    }
+
+    private Model convertBytesToModel(byte[] bytes) {
```

Comment:
This method should not be moved to ProcessModelConverter class ?

Reply:
This is just to cover the Exception to a proper one.